### ISSUE TITLE
to_string() appends 'undef' to array attribute

### ISSUE THREAD

**Issue by jimvella** [dex4er/perl-Exception-Base#3]
Invoking to_string has the side effect of appending 'undef' to an array attribute. Observed in the EPEL package perl-Exception-Base-0.2500-1.el7.noarch.rpm, perl v5.16.3.

Can reproduce with this script:

```
#!/bin/perl
use warnings;
use strict;

use Data::Dumper;

use Exception::Base
    'MyException', => {
        message => 'Validation error',
        has     => [ qw(class errors) ],

        string_attributes => [ 'message', 'class', 'errors' ]
    };
eval {
    MyException->throw(
            class  => __PACKAGE__,
            errors => ["error 1", "error 2", "error 3"]
        );
};

my $exception = $@;

print( Dumper($exception->errors()) . "\n");

$exception->to_string();

print( Dumper($exception->errors()) . "\n");
```

**Comment by dex4er** [comment]
Duplicate of issue #1

**Comment by pghmcfc** [comment]
Fix for #1 does not fix this for me.

**Comment by pghmcfc** [comment]
My apologies, this is indeed fixed in 0.2501.
